lib/hsa/hc_profile.cpp

```diff
@@ -0,0 +1,40 @@
+#include "hc_profile.hpp"
+
+#ifdef CODEXL_PROFILING_ENABLED
+#include "CXLActivityLogger.h"
+#endif
+
+cxlProfiler::cxlProfiler() {
+#ifdef CODEXL_PROFILING_ENABLED
+    if (getenv("HCC_CODEXL_PROFILING")) {
+        enabled = true;
+        cxlMarker::enabled = true;
+        amdtInitializeActivityLogger();
+    }
+#endif
+}
+cxlProfiler::~cxlProfiler() {
+#ifdef CODEXL_PROFILING_ENABLED
+    if (enabled) {
+        amdtFinalizeActivityLogger();
+    }
+#endif
+}
+
+bool cxlMarker::enabled = false;
```

Comment:
@otauqir can we try remove this field, so `hc_profile.cpp` is not needed at all, and we can place all the implementation inside `hc_profile.hpp`? for example, reading from an env var.